qemu/tests/hotplug_mem.py

```diff
@@ -56,35 +63,50 @@ def get_mem_by_name(self, vm, name):
         devices = filter(None, map(vm.devices.get_by_qid, dev_ids))
         return [_[0] for _ in devices]
 
+    def unplug_memory(self, vm, target_mem):
+        """Unplug the target memory, if the memory not exists,
+           hotplug it, then unplug it
+        """
+        devs = self.get_mem_by_name(vm, target_mem)
+        if not devs and self.params.get("strict_check") != "yes":
+            self.hotplug_memory(vm, target_mem)
+        return super(MemoryHotplugSimple, self).unplug_memory(vm, target_mem)
+
     def start_test(self):
         operation = self.params["operation"]
         target_mem = self.params["target_mem"]
+        stage = self.params.get("stage", "before")
+        sub_test_runner = (
+            stage == 'during' and [
+                self.run_background_test] or [
+                self.run_sub_test])[0]
+        func = getattr(self, "%s_memory" % operation)
+        if not callable(func):
+            raise exceptions.TestError(
+                "Unsupported memory operation '%s'" %
+                operation)
+        vm = self.env.get_vm(self.params["main_vm"])
         try:
+            if stage != "after":
+                sub_test = sub_test_runner()
+                func(vm, target_mem)
+                self.check_memory(vm)
```

Comment:
we need prepare subtest before hotplug/unplug mem device both before and during, so I have such if else branch. Maybe I need a doc string here.

Thanks your comments.